### PR TITLE
feat(function): switch to function, add renovate, bump providers, add…

### PR DESCRIPTION
**Pull request by haarchri**
… readme

### Description of your changes
- add renovate
- bump uptest
- bump up
- bump configurations
- add readme

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
=== CONT  kuttl
    harness.go:402: run tests finished
    harness.go:511: cleaning up
    harness.go:553: skipping cluster tear down
    harness.go:554: to connect to the cluster, run: export KUBECONFIG="/Users/haarchri/Documents/ad/configuration-aws-eks-irsa/kubeconfig"
--- PASS: kuttl (2430.19s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (2429.70s)
PASS
13:13:57 [ OK ] running automated tests
```

### CODE REVIEW

File: images/irsa-01.png
Comment (ytsarev) [inline images/irsa-01.png:1]: nit: can we do the same simple diagrams with Mermaid so the are clearly auditable in git?

Comment (haarchri) [reply images/irsa-01.png:1]: we can do this in a new PR - or we can remove the Readme for now - need a new release for some other PRs in pipeline ;)

**Review by ytsarev — APPROVED** [review]
Great stuff, the new doc is very helpful

**Pull request merged**